pkg/utils/types.go

```diff
@@ -0,0 +1,6 @@
+package utils
+
+type ErrorOr[T any] struct {

```

Comment:
I'd prefer we have the types with its own packages, it really clashes with the generic "utils"

Reply:
To clarify, I think we're talking about moving this down to the new `pkg/concurrency` mentioned below? 


Not sure this warrants an entire pkg/types yet